packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -69,6 +69,7 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
     PointerDeviceKind? kind,
     this.dragStartBehavior = DragStartBehavior.start,
     this.velocityTrackerBuilder = _defaultBuilder,
+    this.supportedDevices = _kAllPointerDeviceKinds
```

Comment:
It's weird that that there's a `kind` argument here too.  You should document the interplay between these two (that either can be used to disallow a pointer type).  Maybe even use an example, like what happens if they specify:

```dart
kind: PointerDeviceKind.touch,
supportedDevices: <PointerDeviceKind>{PointerDeviceKind.mouse},
```

Might also make sense to add an assert that they're not specifying a combo that doesn't make any sense.

You could even try to change GestureRecognizer.kind to be "kinds" and a Set, but then we're in breaking change territory.

Reply:
Kinds is definitely odd. If I had the chance I would probably re-do how we do touch slop entirely.

Reply:
Having `kinds` and `supportedDevices` on here is odd. The docs for `kind`:

> It's possible to limit this recognizer to a specific PointerDeviceKind by providing the optional kind argument. If kind is null, the recognizer will accept pointer events from all device kinds.

and `supportedDevices` also read pretty similar in what they do. A little more documentation contrasting the two is a good idea.

Reply:
I am wondering whether we should deprecate `kind` altogether now and redirect people to `supportedDevices` to simplify the API.

Reply:
That's what I was thinking, we should deprecate `kind`. I can write a dart fix and a migration guide to help. :) I already wrote a [really similar one](https://flutter.dev/docs/release/breaking-changes/default-desktop-scrollbars) for some other recent changes to `ScrollBehavior`.

Reply:
Touched base with @jonahwilliams. I'll open a follow up change to see what this will look like. :)

Reply:
FYI @renancaraujo

Reply:
Follow up is in https://github.com/flutter/flutter/pull/81858

Reply:
Thanks @tvolkert for the FYI.

Reply:
Just coming back around to close the loop on this, https://github.com/flutter/website/pull/5771 has a migration guide for those seeking to restore the prior behavior. :)